### ISSUE TITLE
[SP4] WiFi Won't Connect

### ISSUE THREAD

**Issue by SuperRoboWizard** [jakeday/linux-surface#332]
Trying to connect to WiFi results in an endlessly spinning circle in the UI, and repeated batches of the following message in dmesg:
```
mwifiex_pcie: info: trying to associate to '[ssid]' [bssid]
mwifiex_pcie: ASSOC_RESP: failed,   status code=2 err=0xfffc a_id=0x3fff
mwifiex_pcie: assoc failure: reason CONNECT_ERR_ASSOC_ERR_TIMEOUT
mwifiex_pcie: ASSOC_RESP: AUTH timeout
mwifiex_pcie: info: association to bssid [bssid] failed
```
If it's relevant, I'm on OpenSuse Tumbleweed, with the 4.18.20 kernel built from source of course, and the firmware files manually copied. I have yet to notice any functional differences between the Surface kernel and the stock 4.19.11 kernel, including touchscreen functionality, which I will submit a second issue for if it is not related.

**Comment by dondon00** [comment]
Same here. It's asking for the password all the time, and tell me it's not correct. Using Surface Go. Just manual installed the 4.18.20-2 version and WiFi worked again.

**Comment by SuperRoboWizard** [comment]
Just finished compiling and installing 4.18.20-2, and it still doesn't work for me. No WiFi, no touch.

**Comment by jakeday** [comment]
Are you saying that my prebuilt 4.18.20-2 works and 4.18.20-3 does not?

**Comment by SuperRoboWizard** [comment]
I can't use the prebuilt kernel, since I'm on OpenSUSE. I have tried rhe -1 and -2 versions, neither of which worked. I have yet to try the -3 version. It will take a while to compile, so I probably won't have an answer for you for that one until tomorrow.

**Comment by SavellM** [comment]
WiFi on my surface go is too continually asking for password. Saying it's incorrect when it isnt. 
Any specific workaround?

**Comment by xgenvn** [comment]
Using the -3 release but wifi drops randomly, I'm switching to latest mainline kernel and wifi seems a lot better. Need to drop touch support for now.

**Comment by RussH** [comment]
you may want to look here - https://bugs.archlinux.org/task/59052, seems to be a regression in latest kernel

**Comment by RussH** [comment]
Wifi issues - this seems to be a Marvell driver issue, as it looks like our issue is being discussed in bugzilla.kernel.org here; https://bugzilla.kernel.org/show_bug.cgi?id=109681. Seems to be in need of some folks contributing debug logs etc.

**Comment by jakeday** [comment]
This should be fixed in 4.18.20-5, as well as using the updated marvell firmware I pushed. You can also try the new mwlwifi module by blacklisting mwifiex and mwifiex_pcie.
